app/src/main/java/eu/kanade/tachiyomi/util/chapter/ChapterTrackSync.kt

```diff
@@ -40,3 +53,72 @@ fun syncChaptersWithTrackServiceTwoWay(db: DatabaseHelper, chapters: List<Chapte
         }
     }
 }
+
+private var job: Job? = null
+private var oldMangaId: Long? = null
+
+/**
+ * Starts the service that updates the last chapter read in sync services. This operation
+ * will run in a background thread and errors are ignored.
+ */
+fun Controller.updateTrackChapterMarkedAsRead(
+    db: DatabaseHelper,
+    preferences: PreferencesHelper,
+    oldLastChapter: Chapter?,
+    newLastChapter: Chapter?,
+    mangaId: Long?,
+    delay: Long = 3000
+) {
+    if (!preferences.trackMarkedAsRead()) return
+
+    val oldChapterRead = oldLastChapter?.chapter_number?.toInt() ?: 0
+    val newChapterRead = newLastChapter?.chapter_number?.toInt() ?: 0
+
+    // To avoid unnecessary calls if multiple marked as read for same manga
+    if (mangaId == oldMangaId) job?.cancel() else oldMangaId = mangaId
+
+    // We want these to execute even if the presenter is destroyed
+    job = (activity as AppCompatActivity).lifecycleScope.launchIO {
+        delay(delay)
+        updateTrackChapterRead(db, preferences, mangaId, oldChapterRead, newChapterRead)
+        (router.backstack.lastOrNull()?.controller as? MangaDetailsController)?.presenter?.fetchTracks()
+    }
+}
+
+suspend fun updateTrackChapterRead(
+    db: DatabaseHelper,
+    preferences: PreferencesHelper,
+    mangaId: Long?,
+    oldChapterRead: Int,
+    newChapterRead: Int
+) {
+    val trackManager = Injekt.get<TrackManager>()
+    val trackList = db.getTracks(mangaId).executeAsBlocking()
+    trackList.map { track ->
+        val service = trackManager.getService(track.sync_id)
+        if (service != null && service.isLogged) {
+            val shouldCustomCount = listOf(
+                abs(track.last_chapter_read - oldChapterRead), oldChapterRead, track.last_chapter_read
+            ).all { it > 15 }
+            val newCountChapter = if (shouldCustomCount) {
+                (track.last_chapter_read + (newChapterRead - oldChapterRead)).coerceAtLeast(0)
+            } else newChapterRead
+            if (!preferences.context.isOnline()) {
+                val trackings = preferences.trackingsToAddOnline().get().toMutableSet()
+                val currentTracking = trackings.find { it.startsWith("$mangaId:${track.sync_id}:") }
+                trackings.remove(currentTracking)
+                trackings.add("$mangaId:${track.sync_id}:$newCountChapter")
+                preferences.trackingsToAddOnline().set(trackings)
+                DelayedTrackingUpdateJob.setupTask(preferences.context)
+            } else {
+                try {
+                    track.last_chapter_read = newCountChapter
+                    service.update(track, true)
+                    db.insertTrack(track).executeAsBlocking()
+                } catch (e: Exception) {
+                    Timber.e(e)
+                }
+            }
```

Comment:
Well this is almost like how it was before my changes. There is two reasons why I added the old chapter (as far as I remember, did that a few months ago): 
- Allow to decrease last chapter number (except for reader)
- Allow correct tracking for cases like seasons. For example if a manga source has all chapters but the tracker split it into seasons then the tracked value would be incorrect.
Hence this piece of code (the 15 value is random, I thought that a difference of more than 15 chapters between tracker last read and app last read would mean either a season or purposefully different), to increase the tracker chapter by the same difference between old and new app chapters :
```
val shouldCustomCount = listOf(
                abs(track.last_chapter_read - oldChapterRead), oldChapterRead, track.last_chapter_read
            ).all { it > 15 }
            val newCountChapter = if (shouldCustomCount) {
                (track.last_chapter_read + (newChapterRead - oldChapterRead)).coerceAtLeast(0)
            } else newChapterRead
```

Reply:
I think it's overkill to support decrease in chapters read, since it's rarely a case where users are going backwards in progress
For point 2 I'm not sure if you mean the source has all the chapters combined into seasons or the tracker lists the chapters as seasons. If it's the latter that tracker needs to be reported for incorrect formatting. Just about every tracking service supports volumes (seasons) as a separate value, it's just not something that's updated on tachi.

So yeah I think going back to the basics is overall better

Reply:
Yeah the chapter's decrease might not be necessary
About point 2 here is a concrete example: https://anilist.co/manga/129705/Chakage-Salja-Season-1/
Anilist decided to split this manhwa between seasons (why ? no idea), so if a source list all the chapters at once, the last read chapter will be wrong. Like chapter 67 in tachi should be 16 in anilist season 2 (with season 1 marked as completed), but the current feature would force the tracker to put 67 on season 2
I encountered this case a few times (sometimes reversed) but not sure how common it is

Reply:
Well thats annoying, but honestly I'd just leave this be, and tackle it another day. This pr is really clean right now as is so I'd prefer leaving it out for now

Reply:
Okay, fine with me